test/multitenancy/loginMethods.test.js

```diff
@@ -0,0 +1,314 @@
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+const { printPath, setupST, startSTWithMultitenancy, killAllST, cleanST } = require("../utils");
+let assert = require("assert");
+const express = require("express");
+const request = require("supertest");
+let { Querier } = require("../../lib/build/querier");
+let { ProcessState } = require("../../lib/build/processState");
+let SuperTokens = require("../../");
+let Multitenancy = require("../../recipe/multitenancy");
+let EmailPassword = require("../../recipe/emailpassword");
+let ThirdPartyEmaillPassword = require("../../recipe/thirdpartyemailpassword");
+let Passwordless = require("../../recipe/passwordless");
+let MultifactorAuth = require("../../recipe/multifactorauth");
+let Session = require("../../recipe/session");
+let { middleware, errorHandler } = require("../../framework/express");
+
+describe(`loginMethods: ${printPath("[test/multitenancy/loginMethods.test.js]")}`, function () {
+    beforeEach(async function () {
+        await killAllST();
+        await setupST();
+        ProcessState.getInstance().reset();
+    });
+
+    after(async function () {
+        await killAllST();
+        await cleanST();
+    });
+
+    it("test without mfa - case 1", async function () {
+        const connectionURI = await startSTWithMultitenancy();
+        SuperTokens.init({
+            supertokens: {
+                connectionURI,
+            },
+            appInfo: {
+                apiDomain: "api.supertokens.io",
+                appName: "SuperTokens",
+                websiteDomain: "supertokens.io",
+            },
+            recipeList: [
+                EmailPassword.init(),
+                Passwordless.init({
+                    contactMethod: "EMAIL",
+                    flowType: "USER_INPUT_CODE_AND_MAGIC_LINK",
+                }),
+                Multitenancy.init(),
+            ],
+        });
+
+        const app = express();
+
+        app.use(middleware());
+        app.use(errorHandler());
+
+        let response = await new Promise((resolve) =>
+            request(app)
+                .get("/auth/loginmethods")
+                .send()
+                .end((err, res) => {
+                    if (err) {
+                        resolve(undefined);
+                    } else {
+                        resolve(res);
+                    }
+                })
+        );
+
+        assert.deepEqual(response.body.firstFactors, ["emailpassword", "otp-email", "link-email"]);
```

Comment:
Is the array order guaranteed to be the consistent? If we do not care about it you could compare them as sets.

Reply:
it's kind of consistent considering the way it's implemented, changed it to compare like sets anyway